frontend/app-development/features/dataModelling/SchemaEditorWithToolbar/SelectedSchemaEditor.tsx

```diff
@@ -70,7 +79,21 @@ const SchemaEditorWithDebounce = ({ jsonSchema, modelPath }: SchemaEditorWithDeb
 
   useOnUnmount(() => {
     clearTimeout(saveTimeoutRef.current);
-    saveFunction();
+    const jsonModels: DatamodelMetadataJson[] = queryClient.getQueryData([
+      QueryKey.DatamodelsJson,
+      org,
+      app,
+    ]);
+    const xsdModels: DatamodelMetadataXsd[] = queryClient.getQueryData([
+      QueryKey.DatamodelsXsd,
+      org,
+      app,
+    ]);
+    const metadataList = mergeJsonAndXsdData(jsonModels, xsdModels);
+    const datamodelExists = metadataList.some(
+      (datamodel) => datamodel.repositoryRelativeUrl === modelPath,
+    );
```

Comment:
Har du vurdert å flytte dette ut i en egen funksjon for å gjøre koden mer lesbar?
```
  const doesModelExist = useCallback(() => {
    const jsonModels: DatamodelMetadataJson[] = queryClient.getQueryData([
      QueryKey.DatamodelsJson,
      org,
      app,
    ]);
    const xsdModels: DatamodelMetadataXsd[] = queryClient.getQueryData([
      QueryKey.DatamodelsXsd,
      org,
      app,
    ]);
    const metadataList = mergeJsonAndXsdData(jsonModels, xsdModels);
    return metadataList.some(
      (datamodel) => datamodel.repositoryRelativeUrl === modelPath,
    );
  }, [queryClient, modelPath, org, app]);
```